api/v1/routes/volume.ts

```diff
@@ -9,13 +12,72 @@ router.get('/total/tx_count/:direction',
     ],
     async (req: Request, res: Response) => {
 
-    const errors = validationResult(req)
-    if (!errors.isEmpty()) {
-        return res.status(400).json({ errors: errors.array() })
-    }
+        const errors = validationResult(req)
+        if (!errors.isEmpty()) {
+            return res.status(400).json({errors: errors.array()})
+        }
 
-    return res.json({"data":{"ethereum" : []}})
+        let supportedChainIds = getChainIdNums()
+        let direction = req.params.direction;
 
-});
+        let resObj = {
+            "data": {}
+        }
+
+        let collection = await MongoConnection.getBridgeTransactionsCollection()
+        for (const chainId of supportedChainIds) {
+
+            let matchFilter = direction === 'out' ?
+                {'fromChainId': chainId, 'sentTime': {$exists: true}} :
+                {'toChainId': chainId, 'receivedTime': {$exists: true}}
+            let dateField = direction === 'out' ? '$sentTime' : '$receivedTime'
+
+            let txnCountByDate: Document[] = await collection.aggregate([
+                {
+                    $match: matchFilter
+                },
+                {
+                    $addFields: {
+                        'date': {
+                            $dateToString: {
+                                format: "%Y-%m-%d", date: {
+                                    $toDate: {$multiply: [dateField, 1000]}
+                                }
+                            }
+                        }
+                    }
+                },
+                {
+                    $group: {
+                        _id: "$date",
+                        count: {$sum: 1},
+                    }
+                },
+                {
+                    $project: {
+                        _id: 1,
+                        count: {$toInt: "$count"},
+                    }
+                },
+                {
+                    $sort: {
+                        _id: 1
+                    }
+                }
+            ]).toArray()
+
+            // Maps date to count
+            let chainName = getChainNameFromId(chainId)
+            // @ts-ignore
```

Comment:
Multiple ts-ignores, nicer way or am I doing something wrong ?

Reply:
see above, chainname could be undefined in the line above.

There aren't many circumstances where ts-ignore should be used